<launch>
  <!-- Camera driver -->
  <node name="usb_cam" pkg="usb_cam" type="usb_cam_node" output="screen" >
    <param name="video_device" value="/dev/video0" />
    <param name="image_width" value="640" />
    <param name="image_height" value="480" />
    <param name="pixel_format" value="yuyv" />
    <param name="camera_frame_id" value="usb_cam" />
    <param name="io_method" value="mmap"/>
  </node>
  
  <!-- C++ AprilTag detector and pose publisher -->
  <node name="tag_detector" pkg="picopter" type="tag_detector" output="screen"/>
  
  <!-- Python IMU data publisher -->
  <node name="imu_data" pkg="picopter" type="imu_data.py"/>
  
  <!-- Python state estimator -->
  <node name="quad_state" pkg="picopter" type="quad_state.py" output="screen"/>
  
  <!-- Python ESC controller -->
  <node name="motor_control" pkg="picopter" type="motor_control.py"
</launch>
